### PR TITLE
Add architecture decision records

### PR DESCRIPTION
**Pull request by emgsilva**
May be interesting to have "decision records" (like in ADR) for the "significant decisions" taken on the changes on the template (for example, why are we adding a certain element or removing other... this is very helpful to track and understand the evolution of the template/model and the motivations for changes). This may actually be a good addition to all the templates under ddd-crew.

### CODE REVIEW

File: docs/adr/0-template_adr.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ADR template
+
+Template based on [Documenting architecture decisions - Michael Nygard](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions), [Markdown version](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md) from Joel Parker Henderson.
```
Comment (selketjah) [inline docs/adr/0-template_adr.md:3]: The blogpost link is not working for me

Comment (emgsilva) [reply docs/adr/0-template_adr.md:3]: It does open, eventually... takes really long time :( But that is "the reference" on ADRs (and the one used for this particular format/template). We may also take that whole intro part out, or just leave the link to the Markup template I used. What do you think?

Comment (cakper) [reply docs/adr/0-template_adr.md:3]: I usually use archive.org link: https://web.archive.org/web/20200129074448/http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions works better than the original one

Regarding the use of ADRs - I'm in favour - will provide much-needed context :)

Comment (emgsilva) [reply docs/adr/0-template_adr.md:3]: We should ask Michael to upgrade his setup :p Anyways, I think staying with ADRs seems the way to go. I can replace the link so it is "responds in reasonable time" for now and as we move forward we can evaluate if we keep this intro or not on the template. Are you OK with that?

**Comment by emgsilva** [comment]
Added the ```archive.org``` link to Michael Nygard reference work and made some textual improvements on the Template intro (especially to have a sort of clear file name strategy). Take a look and if you are OK, merge it. I would also propose to add the very same template to all the other repos.

**Comment by cakper** [comment]
@emgsilva thank you for your contribution!

**Pull request merged**